runtime/src/main/cpp/Worker.cpp

```diff
@@ -283,6 +286,7 @@ class State {
         workers_.erase(it);
       }
     }
+    GC_UnregisterWorker(worker);
```

Comment:
So the worker gets unregistered when it is already invalid.
Doesn't probably matter with current "registry" implementation, but doesn't look correct semantically anyway.
Also it is not symmetrical to `addWorkerUnlocked` which registers the worker in GC only after it becomes fully valid.

Reply:
What do you mean by `invalid`? Object is still correct, it just cannot be discovered by id.